scripts/mods/StreamingInfo/StreamingInfo.lua

```diff
@@ -157,9 +157,14 @@ mod:hook_safe(GameModeManager, "server_update", function(self)
 end)
 
 mod:hook_safe(IngameUI, "update", function(self, dt, t, disable_ingame_ui, end_of_level_ui) -- luacheck: no unused
-	local level_transition_handler = Managers.state.game_mode.level_transition_handler
+	local level_transition_handler = Managers.level_transition_handler
 	local level_key = level_transition_handler:get_current_level_keys()
-	local is_in_inn = level_key == "inn_level"
+	if not level_key then return end
+
+	local level_settings = LevelSettings[level_key]
+	if not level_settings then return end
+
+	local is_in_inn = level_settings.game_mode == "inn"

```

Comment:
Heads up that this might not show in the CW keep (`"inn_deus`"), although only talents are relevant there.

A simple way to check for the inn that works on all current (and future) inns is:

```lua
local is_in_inn = DamageUtils.is_in_inn
```

Reply:
@ManuelBlanc This mod has a bug [(see comments in mod page)](https://steamcommunity.com/sharedfiles/filedetails/?id=1504702573) where it doesn't work if the keep has been modified by an active event. It can't seem to detect the status of being in the keep when there is an active event. Do you reckon the above fix could solve the issue?